### PR TITLE
chore(flame_steering_behaviors): bring `flame_steering_behaviors` into the repository

### LINKED ISSUE #49: chore: migrate `flame_steering_behaviors` to `flame_behaviors`
**Description**

As noted in https://github.com/VeryGoodOpenSource/flame_steering_behaviors/issues/12 we closed down the `flame_steering_behaviors` repo in favor of moving that over to this repo. This should allow for one cohesive repo with the needed functionallity.

**Requirements**

- [x] The functionality from `flame_steering_behaviors` is fulled moved over to `flame_behaviors`

### PR DESCRIPTION
**Pull request by wolfenrain**
## Status

**READY**

## Description

Closes #49 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

File: .github/workflows/flame_steering_behaviors.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - "packages/flame_steering_behaviors/test/**"
+      - "packages/flame_steering_behaviors/pubspec.yaml"
+
+jobs:
+  build:
+    uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
+    with:
+      working_directory: packages/flame_steering_behaviors
+      flutter_channel: stable
+      flutter_version: 3.13.5
```
Comment (alestiago) [inline .github/workflows/flame_steering_behaviors.yaml:25]: 👀 shouldn't it be Flutter 3.16?

File: packages/flame_steering_behaviors/example/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Example
```
Comment (alestiago) [inline packages/flame_steering_behaviors/example/README.md:1]:
```suggestion
# Flame Steering Behaviors' example
```

File: packages/flame_steering_behaviors/example/analysis_options.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+include: package:very_good_analysis/analysis_options.3.0.1.yaml
```
Comment (alestiago) [inline packages/flame_steering_behaviors/example/analysis_options.yaml:1]: Are we planning to bump this to 5.1.0?

File: packages/flame_steering_behaviors/example/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: example
```
Comment (alestiago) [inline packages/flame_steering_behaviors/example/pubspec.yaml:1]:
```suggestion
name: flame_steering_behaviors_example
```

File: packages/flame_steering_behaviors/test/src/behaviors/wander_behavior_test.dart [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../helpers/helpers.dart';
+
+class _MockRandom extends Mock implements Random {}
+
+const startAngle = 90 * degrees2Radians;
+const randomValue = 0.25;
+const maximumAngle = 90 * degrees2Radians;
```
Comment (alestiago) [inline packages/flame_steering_behaviors/test/src/behaviors/wander_behavior_test.dart:15]: Should these be private?

**Review by alestiago — DISMISSED** [review]
Left some minor comments. Thanks for taking care of this!

I tried the example and it run successfully. If there is something in particular I should take a look at let me know 🙌

File: packages/flame_steering_behaviors/example/pubspec.yaml (hunk 4 above)
Comment (wolfenrain) [reply packages/flame_steering_behaviors/example/pubspec.yaml:1]: Keeping it as `example` as that is what the other example was as well, can always be changed in a follow-up PR

File: packages/flame_steering_behaviors/example/README.md (hunk 2 above)
Comment (wolfenrain) [reply packages/flame_steering_behaviors/example/README.md:1]: Same as with the pubspec name, keeping it as is to keep it in-line with the `flame_behaviors` example

File: .github/workflows/flame_steering_behaviors.yaml (hunk 1 above)
Comment (wolfenrain) [reply .github/workflows/flame_steering_behaviors.yaml:25]: I took it from the `flame_behaviors.yaml` we should update this in a different PR imho

Comment (alestiago) [reply .github/workflows/flame_steering_behaviors.yaml:25]: Okay with me, before resolving the conversation can you link to an issue that tracks this?

Comment (wolfenrain) [reply .github/workflows/flame_steering_behaviors.yaml:25]: Tracked it here: #52

**Review by alestiago — APPROVED** [review]

**Pull request merged**